simtools/utils/names.py

```diff
@@ -18,39 +22,33 @@
     "validate_telescope_name",
 ]
 
-# Telescopes and other array elements
-array_element_names = {
-    # CTAO telescopes
-    "LSTN": {"site": "North", "observatory": "CTAO", "collection": "telescopes"},
-    "MSTN": {"site": "North", "observatory": "CTAO", "collection": "telescopes"},
-    "LSTS": {"site": "South", "observatory": "CTAO", "collection": "telescopes"},
-    "MSTS": {"site": "South", "observatory": "CTAO", "collection": "telescopes"},
-    "SSTS": {"site": "South", "observatory": "CTAO", "collection": "telescopes"},
-    "SCTS": {"site": "South", "observatory": "CTAO", "collection": "telescopes"},
-    # calibration devices
-    "ILLN": {"site": "North", "observatory": "CTAO", "collection": "calibration_devices"},
-    "RLDN": {"site": "North", "observatory": "CTAO", "collection": "calibration_devices"},
-    "STPN": {"site": "North", "observatory": "CTAO", "collection": "calibration_devices"},
-    "MSPN": {"site": "North", "observatory": "CTAO", "collection": "calibration_devices"},
-    "CEIN": {"site": "North", "observatory": "CTAO", "collection": "calibration_devices"},
-    "WSTN": {"site": "North", "observatory": "CTAO", "collection": "calibration_devices"},
-    "ASCN": {"site": "North", "observatory": "CTAO", "collection": "calibration_devices"},
-    "DUSN": {"site": "North", "observatory": "CTAO", "collection": "calibration_devices"},
-    "LISN": {"site": "North", "observatory": "CTAO", "collection": "calibration_devices"},
-    "ILLS": {"site": "South", "observatory": "CTAO", "collection": "calibration_devices"},
-    "RLDS": {"site": "South", "observatory": "CTAO", "collection": "calibration_devices"},
-    "STPS": {"site": "South", "observatory": "CTAO", "collection": "calibration_devices"},
-    "MSPS": {"site": "South", "observatory": "CTAO", "collection": "calibration_devices"},
-    "CEIS": {"site": "South", "observatory": "CTAO", "collection": "calibration_devices"},
-    "WSTS": {"site": "South", "observatory": "CTAO", "collection": "calibration_devices"},
-    "ASCS": {"site": "South", "observatory": "CTAO", "collection": "calibration_devices"},
-    "DUSS": {"site": "South", "observatory": "CTAO", "collection": "calibration_devices"},
-    "LISS": {"site": "South", "observatory": "CTAO", "collection": "calibration_devices"},
-    # other telescopes
-    "MAGIC": {"site": "North", "observatory": "MAGIC", "collection": "telescopes"},
-    "VERITAS": {"site": "North", "observatory": "VERITAS", "collection": "telescopes"},
-    "HESS": {"site": "South", "observatory": "HESS", "collection": "telescopes"},
-}
+
+@lru_cache(maxsize=None)
+def load_array_elements():
+    """
+    Load array elements from reference files and keep in cache.
+
+    Returns
+    -------
+    dict
+        Array elements.
+    """
+    base_path = Path(__file__).parent
+    with open(base_path / "../schemas/array_elements.yml", "r", encoding="utf-8") as file:
+        return yaml.safe_load(file)["data"]
+
+
+def array_elements():
```

Comment:
What is the reason for this function and not calling directly `load_array_elements()`?

Reply:
Good point, this was a leftover from an earlier version.

Fixed this.